src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker_Patterns.cs

```diff
@@ -685,9 +685,10 @@ public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node
             LearnFromAnyNullPatterns(node.Expression, node.Pattern);
             VisitPatternForRewriting(node.Pattern);
             var expressionState = VisitRvalueWithState(node.Expression);
+            bool negated = node.Pattern.IsNegated(out _);
             var labelStateMap = LearnFromDecisionDag(node.Syntax, node.DecisionDag, node.Expression, expressionState, ref this.State);
-            var trueState = labelStateMap.TryGetValue(node.WhenTrueLabel, out var s1) ? s1.state : UnreachableState();
-            var falseState = labelStateMap.TryGetValue(node.WhenFalseLabel, out var s2) ? s2.state : UnreachableState();
+            var trueState = labelStateMap.TryGetValue(negated ? node.WhenFalseLabel : node.WhenTrueLabel, out var s1) ? s1.state : UnreachableState();
```

Comment:
>negated ? node.WhenFalseLabel : node.WhenTrueLabel [](start = 54, length = 50)

A need to do a fixup like that and the need to analyze a child for that feels very fragile. Easy to miss while working with the tree.

Reply:
Yes, it is fragile.

---
In reply to: [530525286](https://github.com/dotnet/roslyn/pull/49369#discussion_r530525286) [](ancestors = 530525286)